recirq/quantum_chess/circuit_transformer.py

```diff
@@ -305,6 +317,384 @@ def transform(self, circuit: cirq.Circuit) -> cirq.Circuit:
+        for i in range(v):
+            if farthest[i] < farthest[center]:
+                center = i
+
+        return index_to_qubit_mapping[center]
+
+    def traverse(
+        self,
+        g: Dict[cirq.Qid, List[Tuple[cirq.Qid, int]]],
+        s: cirq.Qid,
+        visited: Optional[set] = None,
+    ) -> Deque[cirq.Qid]:
+        """Returns a deque of qubits ordered by breadth-first search traversal.
+
+        During each iteration of breadth-first search, the adjacent nodes are
+        sorted by their corresponding moments before being traversed.
+
+        Args:
+          g: A logical qubits graph.
+          s: The source qubit from which to start breadth-first search.
+        """
+        order = deque()
+        if visited is None:
+            visited = set()
+        visited.add(s)
+        queue = deque()
+        queue.append(s)
+        while queue:
+            q = queue.popleft()
+            order.append(q)
+            neighbors_sorted_by_moment = sorted(g[q], key=lambda x: x[1])
+            for neighbor in neighbors_sorted_by_moment:
+                neighbor = neighbor[0]
+                if neighbor not in visited:
+                    visited.add(neighbor)
+                    queue.append(neighbor)
+        return order
+
+    def find_reference_qubits(
+        self,
+        mapping: Dict[cirq.Qid, cirq.GridQubit],
+        lg: Dict[cirq.Qid, List[Tuple[cirq.Qid, int]]],
+        lq: cirq.Qid,
+    ) -> List[cirq.GridQubit]:
+        """Returns a list of physical qubits from which to find the next mapping.
+
+        The nodes adjacent to the logical qubit parameter are sorted by their
+        corresponding moments before being traversed. For each adjacent node
+        that has been mapped to a physical qubit, the mapped physical qubit is
+        added to the result.
+
+        Args:
+          mapping: The current mapping of logical qubits to physical qubits.
+          lg: A logical qubits graph.
+          lq: The logical qubit from which to find reference qubits.
+        """
+        qubits = []
+        neighbors_sorted_by_moment = sorted(lg[lq], key=lambda x: x[1])
+        for neighbor in neighbors_sorted_by_moment:
+            neighbor = neighbor[0]
+            if neighbor in mapping:
+                # This neighbor has been mapped to a physical qubit. Add the
+                # physical qubit to reference qubits.
+                qubits.append(mapping[neighbor])
+        return qubits
+
+    def find_candidate_qubits(
+        self,
+        mapped: ValuesView[cirq.GridQubit],
+        pg: Dict[cirq.GridQubit, List[cirq.GridQubit]],
+        pq: cirq.GridQubit,
+    ) -> List[cirq.GridQubit]:
+        """Returns a list of physical qubits available to be mapped.
+
+        Uses level order traversal until a level with free adjacent node(s) is
+        found.
+
+        Args:
+          mapped: The set of currently mapped physical qubits.
+          lg: A physical qubits graph.
```

Comment:
rename `lg` -> `pg` and `lq` -> `pq` to match the arg names?

Reply:
done!